samples/swift/uidemo/ChatViewController.swift

```diff
@@ -0,0 +1,251 @@
+    guard let name = dict["name"] else { return nil }
+    guard let uid  = dict["uid"]  else { return nil }
+    guard let text = dict["text"] else { return nil }
+    
+    self.name = name
+    self.uid = uid
+    self.text = text
+  }
+}
+
+// View controller demonstrating using a FirebaseCollectionViewDataSource
+// to populate a collection view with chat messages. The relevant code
+// is in the call to `collectionViewDataSource.populateCellWithBlock`.
+//
+// All of the error handling in this controller is done with `fatalError`;
+// please don't copy paste it into your production code.
+class ChatViewController: UIViewController, UICollectionViewDelegateFlowLayout {
+  
+  private static let reuseIdentifier = "ChatCollectionViewCell"
+  
+  @IBOutlet private var collectionView: UICollectionView!
+  @IBOutlet private var textView: UITextView! {
+    didSet {
+      textView.layer.borderColor = UIColor.grayColor().colorWithAlphaComponent(0.5).CGColor
+      textView.layer.borderWidth = 1
+      textView.layer.cornerRadius = 8
+      textView.layer.masksToBounds = true
+    }
+  }
+  @IBOutlet private var sendButton: UIButton!
+  
+  // Used to shift view contents up when the keyboard appears.
+  @IBOutlet weak var bottomConstraint: NSLayoutConstraint!
+  
+  private let auth = FIRAuth.auth()
+  private let chatReference = FIRDatabase.database().reference().child("chats")
+  
+  private var collectionViewDataSource: FirebaseCollectionViewDataSource! = nil
+  
+  private var user: FIRUser?
+  private var query: FIRDatabaseQuery?
+  
+  static func fromStoryboard(storyboard: UIStoryboard = AppDelegate.mainStoryboard) -> ChatViewController {
+    return storyboard.instantiateViewControllerWithIdentifier("ChatViewController") as! ChatViewController
+  }
+  
+  override func viewDidLoad() {
+    super.viewDidLoad()
+    
+    self.collectionView.backgroundColor = UIColor.whiteColor()
+    self.collectionView.delegate = self
+    let layout = self.collectionView.collectionViewLayout as! UICollectionViewFlowLayout
+    layout.minimumInteritemSpacing = CGFloat.max
+    layout.minimumLineSpacing = 4
+    
+    self.auth?.addAuthStateDidChangeListener { (auth, user) in
+      self.user = user
+    }
+    
+    self.sendButton.addTarget(self, action: #selector(didTapSend), forControlEvents: .TouchUpInside)
+  }
+  
+  override func viewDidAppear(animated: Bool) {
+    super.viewDidAppear(animated)
+    
+    self.auth?.signInAnonymouslyWithCompletion { (user, error) in
+      if let error = error {
+        fatalError("Sign in failed: \(error.localizedDescription)")
+      }
+      
+      self.query = self.chatReference.queryLimitedToLast(50)
+      
+      self.collectionViewDataSource = FirebaseCollectionViewDataSource(query: self.query!,
+        prototypeReuseIdentifier: ChatViewController.reuseIdentifier,
+        view: self.collectionView)
+      
+      // The initializer called above--though it takes a collection view--
```

Comment:
This comment is good, but this block in general is pretty long and doing quite a lot of work. Consider adding some more narrative comments so that a reader skimming through the code to pull out examples will have an idea of the context they're copying from.